File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        # Use a map to keep track of remainders and the index where they were found
        # Also use a variable to track the running sum
        # If you found a remainder that is already in the map and there is at least a two index gap between the map's idx and the curr idx, return true
        # This is because a multiple of k was found between the gap
        # Return false at the end of the func
        remainder = {0: -1}
        total = 0
        
        for i in range(len(nums)):
            total += nums[i]
            r = total % k
            
            if r not in remainder:
                remainder[r] = i
            elif i - remainder[r] > 1:
                return True
        
        
        return False

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap<>();
        map.put(0, -1);
        Integer runningSum = 0;
        for(int i = 0; i < nums.length; ++i){
            runningSum += nums[i];
            if(k != 0) runningSum %= k;
            Integer prev = map.get(runningSum);
            if(prev != null){
                if(i - prev > 1) return true;
            } else{
                map.put(runningSum, i);
            }
        }
        
        return false;
    }
}